src/main/java/edu/ucr/cs/riple/taint/ucrtainting/serialization/FixVisitor.java

```diff
@@ -51,10 +52,36 @@ public Void visitConditionalExpression(ConditionalExpressionTree node, Set<Fix>
     return null;
   }
 
+  /**
+   * Visitor for method invocations. For method invocations:
+   *
+   * <ol>
+   *   <li>If return type is not type variable, we annotate the called method.
+   *   <li>If return type is type variable and defined in source code, we annotate the called
+   *       method.
+   *   <li>If return type is type variable and defined in library, we annotate the receiver.
+   * </ol>
+   *
+   * @param node The given tree.
+   * @param fixes The set of fixes to add the fix to.
+   * @return Void null.
+   */
   @Override
   public Void visitMethodInvocation(MethodInvocationTree node, Set<Fix> fixes) {
     if (checker.check(node.getMethodSelect())) {
-      buildFixForElement(node.getMethodSelect());
+      Element element = TreeUtils.elementFromUse(node);
+      if (element == null) {
+        return null;
+      }
+      Symbol.MethodSymbol methodSymbol = (Symbol.MethodSymbol) element;
+      // check for type variable in return type.
+      if (methodSymbol.type.getReturnType().tsym instanceof Symbol.TypeVariableSymbol) {
+        // Build the fix for the receiver.
+        ((MemberSelectTree) node.getMethodSelect()).getExpression().accept(this, fixes);
```

Comment:
I'm not quite sure how the visitor will know that the fix here is to update the type parameter (_not_ to make the receiver itself untainted).  Is that information passed somewhere?

Reply:
@msridhar Yes, that is not implemented in this PR yet, I was planning to address that in a separate PR. But actually now I think it worths working on it in this PR. Will update it now.